app/src/main/java/org/thoughtcrime/securesms/ui/theme/Themes.kt

```diff
@@ -29,15 +31,15 @@ var selectedTheme: ThemeColors? = null
 fun SessionMaterialTheme(
     content: @Composable () -> Unit
 ) {
-    // set the theme data if it hasn't been done yet
-    if(selectedTheme == null) {
-        // Some values can be set from the preferences, and if not should fallback to a default value
-        val context = LocalContext.current
-        val preferences = AppTextSecurePreferences(context)
-        selectedTheme = preferences.getComposeTheme()
-    }
+    val context = LocalContext.current
+    val preferences = AppTextSecurePreferences(context)
+
+    val selectedColorSet = selectedColorSet ?: preferences.getColorSet().also { selectedColorSet = it }
 
-    SessionMaterialTheme(colors = selectedTheme ?: ClassicDark()) { content() }
+    SessionMaterialTheme(
+        colors = selectedColorSet.get(isSystemInDarkTheme()),
```

Comment:
We could maybe move the `isSystemInDarkTheme` check directly inside the `get`?
`ThemeColorSet` is about Compose colors so I think it's ok to make the `get` function a composable and let it have the logic internally? What do you think?

Reply:
Yeah, I like it.

I was also thinking, because I don't like the naming...

- `LightDarkColors` - verbose
- `ColorSet` - 1. it's not a `Set`, 2. `Colors` is kind of a "set of colors"

... that it might even help to go more OO, and have 2 implementations...


```
class StaticColors: LightDarkColors(val colors) {
    override get() = colors
}

class DynamicColors: LightDarkColors(val light, val dark) {
    @Composable get() = isSystemDark() dark else light
}
```

It's a `@Composable () -> Colors` so... `DynamicColors`?

```
typealias DynamicColors = @Composable () -> Colors
fun StaticColors(colors: Colors): DynamicColors = { colors }
fun DynamicColors(light: Colors, dark: Colors): DynamicColors = { if (isSystemDark() dark else light }
```

or something?